### ISSUE TITLE
VideoSync Error for TTS Bluray

### ISSUE THREAD

**Issue by ericchristoffersen** [GoldenCheetah/GoldenCheetah#3702]
Am seeing bad av sync for tts bluray discs. Off by 5.7%.

Debugged and cause is that tts declared fps is 26.454, but video actual fps is 25.0.

Currently seconds are determined at tts load time, translation doesn't know actual video frame rate, only what is declared.

Fix requires us to translate loaded videosync seconds into actual video seconds after video fps is known.

**Comment by ericchristoffersen** [comment]
Ok. This was fun and interesting. First there is the videosync frame rate error, where videosync file's frame rate doesn't match its associated video file. This actually happens on tacx bluray discs, difference is 5-6%.

When I was testing this fix I found a bunch of my favorite non-tacx rides, and some tacx rides, were suffering from an av sync problem that they didn't have before.

Debugging I found that the cases that didn't work were cases where the tts file had no location data, so I was using a separate gpx file as 'workout' file. In those cases the GPX route distance did not match the videosync distance. Previously this error was covered up because the above frame rate issue, but now that frame lookup is correct the distance difference suddenly matters.

I added a second videosyncdistanceadjustmentfactor, to cover the difference between videosync distance and workout distance.

With that second change all my rides now work very well. Even the ones that werent broken now see better videosync.

So... this is all about mitigating differences between files. Since videosync is the only thing that speaks both distance and time its the best place to fix the unit disagreements between workout and media.

To summarize:

For common case: tacx tts for workout, same tacx tts for videosync, tacx video, also json workout, json for videosync:
    - workout distance and videosync distance matches, distance adjustment factor is 1.
    - media framerate and videosync frame rate might differ, videosynctimeadjust is what it needs to be.

For old virtual ride hack case with mismatched gpx and tts:
    - workout distance and videosync distance might be different, adjustment factor is what it needs to be
    - media framerate and videosync frame rate might differ, videosynctimeadjust is what it needs to be.
